src/main/java/fi/aalto/cs/apluscourses/ui/base/TreeView.java

```diff
@@ -35,14 +38,14 @@
       };
 
   private final transient Set<ActionListener> nodeAppliedListeners = ConcurrentHashMap.newKeySet();
+  private final Object popupMenuLock = new Object();
```

Comment:
Should this be transient?

Reply:
It doesn't really matter as we are never gonna serialize these objects.

But logically, yes, the lock would be transient. So you can make it transient to mute Sonar.